index.html

```diff
@@ -194,49 +198,228 @@ <h2>
+                    Once selected, the source of a
+                    <code><a>MediaStreamTrack</a></code> MUST NOT change.</p>
+                    <p>User Agents are encouraged to warn users against sharing
+                    <a>browser</a> display devices as well as <a>monitor</a>
+                    display devices where browser windows are visible, or
+                    otherwise try to discourage their selection on the basis
+                    that these represent a significantly higher risk when shared.</p>
+                    <p>If the result of the request is "granted", then for
+                    each device that is sourcing the provided media, using
+                    a stable and private id for the device, <var>deviceId</var>,
+                    set [[\devicesLiveMap]]<var>[deviceId]</var> to
+                    <code>true</code>, if it isn’t already <code>true</code>,
+                    and set the
+                    [[\devicesAccessibleMap]]<var>[deviceId]</var> to
+                    <code>true</code>, if it isn’t already
+                    <code>true</code>.</p>
+                    <p>The User Agent MUST NOT
+                    <a href="#create-permission">create a permission storage entry</a>
+                    with a value of "granted".
+                    </p>
+                    <p>If the result is "denied", jump to the step labeled
+                    <em>Permission Failure</em> below. If the user never
+                    responds, this algorithm stalls on this step.</p>
+                    <p>If the user grants permission but a hardware error
+                    such as an OS/program/webpage lock prevents access,
+                    <a>reject</a> <var>p</var> with a new
+                    <code><a>DOMException</a></code> object whose
+                    <code><a>name</a></code> attribute has the value
+                    <code>NotReadableError</code> and abort these steps.</p>
+                    <p>If the result is "granted" but device access fails for
+                    any reason other than those listed above, <a>reject</a>
+                    <var>p</var> with a new <code><a>DOMException</a></code>
+                    object whose <code><a>name</a></code> attribute has the
+                    value <code>AbortError</code> and abort these steps.</p>
+                  </li>
+                  <li>
+                    <p>Let <var>stream</var> be the
+                    <code><a>MediaStream</a></code> object for which the user
+                    granted permission.</p>
+                  </li>
+                  <li>
+                    <p>Run the <a href=
+                    "https://w3c.github.io/mediacapture-main/#dfn-applyconstraints-algorithm">
+                    <dfn>ApplyConstraints algorithm</dfn></a> on all
+                    tracks in <var>stream</var> with the appropriate
+                    constraints. Should this fail, let <var>failedConstraint</var>
```

Comment:
nit: "with the appropriate constraints" should be clarified. As-is, both boolean and constraint set values are allowed (do we need to convert "true" to "{}"?), and it is not explicit about applying "audio" constraints to audio and "video" constraints to video. This may seem obvious but as an algorithm step it is vague.